File: backend/src/main/java/com/netdisk/entity/File.java
package com.netdisk.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 文件实体
 */
@Data
@TableName("files")
public class File {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("file_name")
    private String fileName;
    
    @TableField("original_name")
    private String originalName;
    
    @TableField("file_size")
    private Long fileSize = 0L;
    
    @TableField("content_type")
    private String contentType;
    
    @TableField("storage_path")
    private String storagePath;
    
    @TableField("file_hash")
    private String fileHash;
    
    @TableField("owner_id")
    private Long ownerId;
    
    @TableField("parent_folder_id")
    private Long parentFolderId;
    
    @TableField("is_folder")
    private Boolean isFolder = false;
    
    @TableField("is_favorite")
    private Boolean isFavorite = false;
    
    @TableField("created_at")
    private LocalDateTime createdAt;
    
    @TableField("updated_at")
    private LocalDateTime updatedAt;
    
    @TableField("deleted_at")
    private LocalDateTime deletedAt;
    
    @TableLogic
    private Integer deleted = 0;
}

File: frontend/src/api/file.ts
import api from './index'

export interface FileItem {
  id: number
  fileName: string
  originalName: string
  fileSize: number
  contentType: string
  storagePath: string
  ownerId: number
  parentFolderId: number | null
  isFolder: boolean
  isFavorite: boolean
  createdAt: string
  updatedAt: string
}

export interface FileListResponse {
  items: FileItem[]
  total: number
  page: number
  size: number
}

export const fileApi = {
  // 获取文件列表
  getFileList: (params: { parentId?: number | null; page?: number; size?: number }) => {
    return api.get<FileListResponse>('/files', { params })
  },
  
  // 上传文件
  uploadFile: (file: File, parentId?: number | null) => {
    const formData = new FormData()
    formData.append('file', file)
    if (parentId) {
      formData.append('parentId', parentId.toString())
    }
    return api.post<FileItem>('/files/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    })
  },
  
  // 获取文件信息
  getFile: (id: number) => api.get<FileItem>(`/files/${id}`),
  
  // 下载文件
  downloadFile: (id: number) => {
    return api.get(`/files/${id}/download`, { responseType: 'blob' })
  },
  
  // 删除文件
  deleteFile: (id: number) => api.delete(`/files/${id}`),
  
  // 重命名文件
  renameFile: (id: number, name: string) => {
    return api.put(`/files/${id}`, { name })
  },
  
  // 移动文件
  moveFile: (fileId: number, targetFolderId: number) => {
    return api.post('/files/move', { fileId, targetFolderId })
  },
  
  // 收藏文件
  toggleFavorite: (id: number) => api.post(`/files/${id}/favorite`),
  
  // 搜索文件
  searchFiles: (keyword: string, page = 1, size = 20) => {
    return api.get<FileListResponse>('/files/search', { params: { keyword, page, size } })
  },
  
  // 创建文件夹
  createFolder: (name: string, parentId?: number | null) => {
    return api.post('/folders', { name, parentId })
  },
  
  // 删除文件夹
  deleteFolder: (id: number) => api.delete(`/folders/${id}`),
  
  // 获取回收站
  getTrash: (page = 1, size = 20) => {
    return api.get<FileListResponse>('/trash', { params: { page, size } })
  },
  
  // 恢复文件
  restoreFile: (id: number) => api.post(`/trash/${id}/restore`),
  
  // 彻底删除
  permanentDelete: (id: number) => api.delete(`/trash/${id}`),
  
  // 清空回收站
  emptyTrash: () => api.delete('/trash/empty')
}
